File: web-app/src/lib/features/features/directory-rule.ts
import type { Feature } from '../types';

export const directoryRuleFeatures = {
  'directory-rule-delete': {
    id: 'directory-rule-delete',
    title: 'Delete Directory Rule',
    description: 'Deletes a directory rule by path, removing any associated session configuration for that directory.',
    rpcIds: ['directory-rule:delete'],
    componentPaths: [],
    testIds: [
      'TestDeleteDirectoryRule_NotFound',
      'TestDeleteDirectoryRule_EmptyPath',
      'TestDeleteDirectoryRule_Success',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'directory-rule-upsert': {
    id: 'directory-rule-upsert',
    title: 'Upsert Directory Rule',
    description: 'Creates or updates a directory rule, associating session configuration with a given directory path.',
    rpcIds: ['directory-rule:upsert'],
    componentPaths: [],
    testIds: [
      'TestUpsertDirectoryRule_EmptyPath',
      'TestUpsertDirectoryRule_NilRule',
      'TestUpsertDirectoryRule_ValidPath',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/directory_rule.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

var DirectoryRuleDelete = featureregistry.Feature{
	ID:          "directory-rule-delete",
	Title:       "Delete Directory Rule",
	Description: "Deletes a directory rule by path, removing any associated session configuration for that directory.",
	RPCIDs:      []string{"directory-rule:delete"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

var DirectoryRuleUpsert = featureregistry.Feature{
	ID:          "directory-rule-upsert",
	Title:       "Upsert Directory Rule",
	Description: "Creates or updates a directory rule, associating session configuration with a given directory path.",
	RPCIDs:      []string{"directory-rule:upsert"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(DirectoryRuleDelete)
	featureregistry.Register(DirectoryRuleUpsert)
}
